lib/active_model_serializers/adapter/json_api.rb

```diff
@@ -294,7 +294,7 @@ def attributes_for(serializer, fields)
 
       # {http://jsonapi.org/format/#document-resource-objects Document Resource Objects}
       def resource_object_for(serializer)
-        resource_object = serializer.cache_check(self) do
+        resource_object = serializer.fetch(self) do
```

Comment:
yeah, I have an unpushed branch where I added readthis to the benchmark  :)

Thanks for looking!

didn't quite finish this, obvs isn't quite right:

```diff
diff --git a/Gemfile b/Gemfile
index 654f77a..25aa836 100644
--- a/Gemfile
+++ b/Gemfile
@@ -39,6 +39,11 @@ gem 'tzinfo-data', platforms: (@windows_platforms + [:jruby])
 group :bench do
   # https://github.com/rails-api/active_model_serializers/commit/cb4459580a6f4f37f629bf3185a5224c8624ca76
   gem 'benchmark-ips', require: false, group: :development
+  gem 'oj'
+  gem 'oj_mimic_json'
+  gem 'redis', require: false
+  gem 'readthis', require: false
+  gem 'hiredis', require: false
 end
 
 group :test do
diff --git a/test/benchmark/app.rb b/test/benchmark/app.rb
index bc0d368..21d62d0 100644
--- a/test/benchmark/app.rb
+++ b/test/benchmark/app.rb
@@ -36,7 +36,16 @@ class BenchmarkApp < Rails::Application
   config.cache_classes = true
   # CONFIG: CACHE_ON={on,off}
   config.action_controller.perform_caching = ENV['CACHE_ON'] != 'off'
-  config.action_controller.cache_store = ActiveSupport::Cache.lookup_store(:memory_store)
+  # config.action_controller.cache_store = ActiveSupport::Cache.lookup_store(:memory_store)
+  require 'redis'
+  require 'readthis'
+  ENV['REDIS_URL'] = 'redis://localhost:6379/5'
+  config.action_controller.cache_store = :readthis_store, {
+    expires_in: 2.weeks.to_i,
+    namespace: 'cache',
+    redis: { url: ENV.fetch('REDIS_URL'), driver: :hiredis }
+  }
+
 
   config.active_support.test_order = :random
   config.secret_token = 'S' * 30
diff --git a/test/benchmark/fixtures.rb b/test/benchmark/fixtures.rb
index 5242db2..94c80f9 100644
--- a/test/benchmark/fixtures.rb
+++ b/test/benchmark/fixtures.rb
@@ -48,13 +48,13 @@ end
 Rails.configuration.serializers << CachingAuthorSerializer
 
 class CachingCommentSerializer < CommentSerializer
-  cache expires_in: 1.day, skip_digest: true
+  cache expires_in: 1.day.to_i, skip_digest: true
 end
 Rails.configuration.serializers << CachingCommentSerializer
 
 # see https://github.com/rails-api/active_model_serializers/pull/1690/commits/68715b8f99bc29677e8a47bb3f305f23c077024b#r60344532
 class CachingPostSerializer < ActiveModel::Serializer
-  cache key: 'post', expires_in: 0.1, skip_digest: true
+  cache key: 'post', expires_in: 1.day.to_i, skip_digest: true
 
   attributes :id, :title, :body
 
@@ -89,7 +89,7 @@ Rails.configuration.serializers << CachingCommentSerializer
 
 # see https://github.com/rails-api/active_model_serializers/pull/1690/commits/68715b8f99bc29677e8a47bb3f305f23c077024b#r60344532
 class FragmentCachingPostSerializer < ActiveModel::Serializer
-  cache key: 'post', expires_in: 0.1, skip_digest: true
+  cache key: 'post', expires_in: 1.day.to_i, skip_digest: true
 
   attributes :id, :title, :body

```

and then started looking at knuckles

```diff
diff --git a/lib/active_model_serializers/adapter/attributes.rb b/lib/active_model_serializers/adapter/attributes.rb
index 503f024..8294e91 100644
--- a/lib/active_model_serializers/adapter/attributes.rb
+++ b/lib/active_model_serializers/adapter/attributes.rb
@@ -1,3 +1,18 @@
+# require 'oj'
+require 'readthis'
+require 'knuckles'
+ENV['REDIS_URL'] = 'redis://localhost:6379/5'
+Knuckles.configure do |config|
+  config.cache = Readthis::Cache.new(
+    # marshal: Oj,
+    compress: true,
+    driver: :hiredis,
+    url: ENV.fetch('REDIS_URL')
+  )
+
+  config.keygen = Readthis::Expanders
+  # config.serializer = Oj
+end
 module ActiveModelSerializers
   module Adapter
     class Attributes < Base
@@ -18,11 +33,36 @@ module ActiveModelSerializers
         options = serialization_options(options)
 
         if serializer.respond_to?(:each)
+          objects = [object] # serializer.map(&:object)
+          opts.merge!(
+            view: ActiveModelSerializers::SerializableResource,
+            serializer: self.class
+          )
+          Knuckles::Pipeline.new.call(objects, opts)
           serializable_hash_for_collection(options)
         else
           serializable_hash_for_single_resource(options)
         end
       end
+      # module Knuckles::Stages::Renderer
+      #   def call(objects, options)
+      #     objects.each do |hash|
+      #       unless hash[:cached?]
+      #         hash[:result] = do_render(hash[:object], view, options)
+      #       end
+      #     end
+      #   end
+      #
+      #   private
+      #
+      #   def do_render(object, view, options)
+      #     case view
+      #     when Knuckles::View then view.render(object, options)
+      #     else view.new(object, options).as_json
+      #     end
+      #   end
+      # end
+
 
       private
 
```